File: Builder/PHP/TextBuilder.php
<?php
  class TextBuilder implements Builder {
    private $content;

    public function buildHeader($header) {
      $this->content .= "Text header: " . $header . "\n";
    }

    public function buildBody($body) {
      $this->content .= "Text body: " . $body . "\n";
    }

    public function buildFooter($footer) {
      $this->content .= "Text footer: " . $footer . "\n";
    }

    public function getResult() {
      return $this->content;
    }
  }

File: Builder/Go/builder/TextBuilder.go
package builder

type TextBuilder struct {
  content string
}

func (this *TextBuilder) buildHeader(header string) {
  this.content += "header: " + header + "\n"
}

func (this *TextBuilder) buildBody(body string) {
  this.content += "body: " + body + "\n"
}

func (this *TextBuilder) buildFooter(footer string) {
  this.content += "footer: " + footer + "\n"
}

func (this *TextBuilder) GetResult() string {
  return this.content
}
